solver/src/main.rs

```diff
@@ -201,14 +204,27 @@ async fn main() {
         .expect("failed to create gas price estimator"),
     );
 
+    let current_block_stream =
+        current_block_stream(web3.clone(), args.shared.block_stream_poll_interval_seconds)
+            .await
+            .unwrap();
+
     let pair_providers =
         pool_aggregating::pair_providers(&args.shared.baseline_sources, chain_id, &web3).await;
-
     let pool_aggregator = PoolAggregator::from_providers(&pair_providers, &web3).await;
+    let pool_fetcher = Arc::new(
```

Comment:
Pool fetcher is also used inside `UniswapLikeLiquidity` (https://github.com/gnosis/oba-services/blob/52a710f9bf000a77ff70c1e4c53311382030f59a/solver/src/liquidity/uniswap.rs#L83) for when liquidity_collector uses it. Maybe there is a way to share the cache between the two callsites (might require more refactoring though)?

Reply:
At first glance that it is more involved with refactoring so I am going to to do it in a separate PR.